core/state/journal.go

```diff
@@ -162,7 +163,7 @@ func (ch resetObjectChange) revert(s *StateDB) {
 }
 
 func (ch resetObjectChange) dirtied() *common.Address {
-	return nil
+	return ch.account
```

Comment:
The address field here is not needed because `resetObjectChange` already contains the `stateObject`.

Reply:
For future readers: the semantics behind `dirtied()` is, that it it supposed to spit out which, if any, `address` was made `dirty` by the change. And `dirty` means that it affects the state trie, so needs to update the stateroot.

For example, the `refundChange` do not mark something `dirty`, because it only affects the runtime but has no persistent effect on the state. Same with `accessListAddAccountChange`,`addPreimageChange`, `addLogChange` and `transientStorageChange`